lib/core/presentation/widgets/poap/hot_badge_item.dart

```diff
@@ -3,77 +3,116 @@
 
   @override
   Widget build(BuildContext context) {
     return SizedBox(
       width: _badgeThumbnailHeight,
       height: _badgeCollectionThumbnailHeight * 1.6,
-      child: Stack(
-        children: [
-          const _BadgeThumbnail(),
-          const Align(
-            alignment: Alignment(0, 0.3),
-            child: _BadgeCollectionThumbnail(),
+      child: FutureBuilder(
+        future: getIt<TokenRepository>().getToken(
+          input: GetTokenDetailInput(
+            id: '${badge.contract!}-0'.toLowerCase(),
+            network: badge.network,
           ),
-          const Align(
-            alignment: Alignment.bottomCenter,
-            child: _BadgeName(),
-          ),
-          Positioned(
-            top: Spacing.superExtraSmall,
-            right: Spacing.superExtraSmall,
-            child: const _BadgeLocationTag(),
-          ),
-          Align(
-            alignment: const Alignment(0, 0.3),
-            child: Stack(
-              children: [
-                Transform.flip(
-                  flipY: true,
-                  child: CustomPaint(
-                    painter: _BadgeQuantityBarPainter(),
-                    size: _badgeQuantityBarSize,
-                  ),
+        ),
+        builder: (context, snapshot) {
+          final token = snapshot.data?.fold((l) => null, (token) => token);
+          return Stack(
+            children: [
+              _BadgeThumbnail(tokenMetadata: token?.metadata),
+              Align(
+                alignment: const Alignment(0, 0.3),
+                child: _BadgeCollectionThumbnail(badge: badge),
+              ),
+              Align(
+                alignment: Alignment.bottomCenter,
+                child: _BadgeName(
+                  tokenMetadata: token?.metadata,
+                  badge: badge,
                 ),
-                Transform.flip(
-                  flipY: true,
-                  child: TweenAnimationBuilder(
-                    duration: const Duration(milliseconds: 500),
-                    tween: _tween,
-                    builder: (context, animationValue, _) => CustomPaint(
-                      painter: _BadgeQuantityBarPainter(isGradient: true, progress: animationValue),
-                      size: _badgeQuantityBarSize,
+              ),
+              Positioned(
+                top: Spacing.superExtraSmall,
+                right: Spacing.superExtraSmall,
+                child: _BadgeLocationTag(badge: badge),
+              ),
+              Align(
+                alignment: const Alignment(0, 0.3),
+                child: Stack(
+                  children: [
+                    Transform.flip(
+                      flipY: true,
+                      child: CustomPaint(
+                        painter: _BadgeQuantityBarPainter(),
+                        size: _badgeQuantityBarSize,
+                      ),
                     ),
-                  ),
+                    Transform.flip(
```

Comment:
This transform quite complex,  should we separate to some build widget for more clear and easier comments code logic also

Reply:
👍